src/topology/hom/open.lean

```diff
@@ -0,0 +1,105 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import topology.continuous_function.basic
+
+/-!
+# Continuous open maps
+
+This file defines bundled continuous open maps.
+
+We use the `fun_like` design, so each type of morphisms has a companion typeclass which is meant to
+be satisfied by itself and all stricter types.
+
+## Types of morphisms
+
+* `continuous_open_map`: Continuous open maps.
+
+## Typeclasses
+
+* `continuous_open_map_class`
+-/
+
+open function
+
+variables {F α β γ δ : Type*}
+
+/-- The type of continuous open maps from `α` to `β`, aka Priestley homomorphisms. -/
+structure continuous_open_map (α β : Type*) [topological_space α] [topological_space β]
+  extends continuous_map α β :=
+(map_open' : is_open_map to_fun)
+
+infixr ` →CO `:25 := continuous_open_map
+
+/-- `continuous_open_map_class F α β` states that `F` is a type of continuous open maps.
+
+You should extend this class when you extend `continuous_open_map`. -/
+class continuous_open_map_class (F : Type*) (α β : out_param $ Type*) [topological_space α]
+  [topological_space β] extends continuous_map_class F α β :=
+(map_open (f : F) : is_open_map f)
+
+export continuous_open_map_class (map_open)
+
+instance [topological_space α] [topological_space β] [continuous_open_map_class F α β] :
+  has_coe_t F (α →CO β) :=
+⟨λ f, ⟨f, map_open f⟩⟩
+
+/-! ### Top homomorphisms -/
```

Comment:
I don't understand what this header signifies.

Otherwise, fine!

bors d+

Reply:
You better not! Copy creep.